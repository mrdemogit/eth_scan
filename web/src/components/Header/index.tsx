import React, { Component } from 'react';
import styled from 'styled-components/macro';

const WrapperStyled = styled.div`
  border-bottom: 0.01rem solid #979797;
  padding: 1rem;
`;

const HeaderStyled = styled.header`
  font-size: 2rem;
  padding: 1rem;
`;

const HeaderInfo = styled.div`
  font-size: 1rem;
  font-weight: 100;
`;

interface Props {
  balance: boolean;
}

class Header extends Component<Props> {
  render() {
    const { balance } = this.props;
    return (
      <WrapperStyled>
        <HeaderStyled>My Portfolio</HeaderStyled>
        <HeaderInfo>
          {balance ? '4411eth' : 'Enter an Ethereum address to get started'}
        </HeaderInfo>
      </WrapperStyled>
    );
  }
}

export default Header;
