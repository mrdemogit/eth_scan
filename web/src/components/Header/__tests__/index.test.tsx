import React from 'react';
import { render } from 'react-testing-library';
import Header from '../index';

describe('Header', () => {
  it('Render default header', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('Render custom header', () => {
    const { container } = render(<Header custom={<div>My Header</div>} />);
    expect(container).toMatchSnapshot();
  });
});
