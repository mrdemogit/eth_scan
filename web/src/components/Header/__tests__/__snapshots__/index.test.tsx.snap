// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render custom header 1`] = `
<div>
  <div
    class="Header__WrapperStyled-sc-1e5fucl-0 gqwNmP"
  >
    <header
      class="Header__HeaderStyled-sc-1e5fucl-1 fJqWDQ"
    >
      My Portfolio
    </header>
    <div
      class="Header__HeaderInfo-sc-1e5fucl-2 zCCez"
    >
      <div>
        My Header
      </div>
    </div>
  </div>
</div>
`;

exports[`Header Render default header 1`] = `
<div>
  <div
    class="Header__WrapperStyled-sc-1e5fucl-0 gqwNmP"
  >
    <header
      class="Header__HeaderStyled-sc-1e5fucl-1 fJqWDQ"
    >
      My Portfolio
    </header>
    <div
      class="Header__HeaderInfo-sc-1e5fucl-2 zCCez"
    >
      Enter an Ethereum address to get started
    </div>
  </div>
</div>
`;
